### DISCUSSION TITLE
ustack not symbolicated for PIE executables

### DISCUSSION THREAD

**Discussion by 9cel** [bpftrace/bpftrace#3630] [Q&A]
Basically what the title says... when I compile my program without `-no-pie` I don't get symbol names in my `ustack` output. I read some previous issues and my understanding was that ASLR PIE executables caused some trouble due to the symbol caching mechanism, but even with symbol caching disabled I don't get any symbols.

The program is compiled with `-O0 -g3 -fno-omit-frame-pointer`, and if I add `-no-pie` then I do get symbols.

I will note that I'm using WSL2 in case that may be a part of the issue. I saw that some other users had issues with WSL as well but it seemed in their case that it was related to the traced process running inside of a container. In my case nothing is containerized.

**Comment by ajor** [comment]
`ustack` symbolication is working for me with PIE executables. Have you tried different options for the `cache_user_symbols` config option? https://github.com/bpftrace/bpftrace/blob/master/man/adoc/bpftrace.adoc#cache_user_symbols

One other thing is that bpftrace requires the target process to still be running at symbolication-time in order for this to work. If you have a short-lived program then bpftrace might be symbolicating after the target process has quit (bear in mind that bpftrace's `printf` is asynchronous so can result in this behaviour).
